### DISCUSSION TITLE
Trying to query based off an implicit many-to-many relationship

### DISCUSSION THREAD

**Discussion by StokleyO** [prisma/prisma#22217] [Q&A] [answered]
### Question

I have two models that have a many-to-many relationship with one another.

```prisma
 model User{
  id  String    @id @default(uuid())
  inbox Thread[]
}
  
model Thread {
  id                 String    @id @default(uuid())
  participants User[]
  messages     Message[]
}
```

When a user sends a new message, I have them generate a new Message object, and connect themselves as a sender. However, next, I want to be able to query all my threads to see if any thread exists with a specific list of user participants. If none exists, I want to create one and connect this message. If one does exist, I just want to append this new message to it

**Comment by ludralph** [comment, marked as answer]
Hi @StokleyO 👋 

Thank you for raising this question.
To determine whether a thread exists with a specific set of user participants, you can write code that checks for existing threads as shown below
```js
function findThreadWithParticipants(participants: User[]): Thread | null {
  return prisma.thread.findFirst({
  where: {
    participants: {
      every: {
        id: {
          in: participants.map((user) => user.id),
        },
      },
    },
  },
  include: {
    participants: true,
  },
  });
}
```
Then check if the thread exist before creating a new one as shown below:
```js
export async function createOrUpdateThread(message: Message): Promise<Thread> {
  const participants = [message.sender, message.recipient];
  const existingThread = findThreadWithParticipants(participants);

  if (existingThread) {
    // If the thread exists, append the message to it
    await prisma.thread.update({
      where: { id: existingThread.id },
      data: {
        messages: {
          connect: { id: message.id },
        },
      },
    });

    return existingThread;
  } else {
    // Create a new thread if it doesn't exist
    const newThread = await prisma.thread.create({
      data: {
        participants: {
          connect: participants.map((user) => ({ id: user.id })),
        },
        messages: {
          connect: { id: message.id },
        },
      },
    });

    return newThread;
  }
}
```
You may also want to share your complete schema so I can create something specific based off of your schema.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by StokleyO** [reply]
Thank you! I ended up figuring out I’d have to write logic to check this and did so yesterday. I appreciate your help!!

**Reply by StokleyO** [reply]
@ludralph
